pontoon/base/models.py

```diff
@@ -67,6 +67,28 @@ def validate_cldr(value):
                 '%s must be a list of integers between 0 and 5' % value)
 
 
+def get_latest_activity(translations):
+    """Get latest activity data from provided translations."""
+    if not translations.exists():
+        return
+
+    latest_translation = translations.order_by('-date')[0]
+    latest_approval = translations.order_by('-approved_date')[0]
+
+    if latest_approval.approved_date and latest_translation.date < latest_approval.approved_date:
+        latest = latest_approval
```

Comment:
Might as well just return the object here, makes it easier to see the difference between the approval vs the translation, even though it's slightly more lines of code.